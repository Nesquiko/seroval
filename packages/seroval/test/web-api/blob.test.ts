import { describe, it, expect } from 'vitest';
import 'node-fetch-native/polyfill';
import {
  crossSerializeAsync,
  deserialize,
  fromJSON,
  serializeAsync,
  toJSONAsync,
} from '../../src';

describe('Blob', () => {
  describe('serializeAsync', () => {
    it('supports Blob', async () => {
      const example = new Blob(['Hello World'], {
        type: 'text/plain',
      });
      const result = await serializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
      const back = await deserialize<Promise<Blob>>(result);
      expect(back).toBeInstanceOf(Blob);
      expect(await back.text()).toBe(await example.text());
      expect(back.type).toBe(example.type);
    });
  });
  describe('toJSONAsync', () => {
    it('supports Blob', async () => {
      const example = new Blob(['Hello World'], {
        type: 'text/plain',
      });
      const result = await toJSONAsync(Promise.resolve(example));
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = await fromJSON<Promise<Blob>>(result);
      expect(back).toBeInstanceOf(Blob);
      expect(await back.text()).toBe(await example.text());
      expect(back.type).toBe(example.type);
    });
  });
  describe('crossSerializeAsync', () => {
    it('supports Blob', async () => {
      const example = new Blob(['Hello World'], {
        type: 'text/plain',
      });
      const result = await crossSerializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports Blob', async () => {
        const example = new Blob(['Hello World'], {
          type: 'text/plain',
        });
        const result = await crossSerializeAsync(Promise.resolve(example), { scopeId: 'example' });
        expect(result).toMatchSnapshot();
      });
    });
  });
});
