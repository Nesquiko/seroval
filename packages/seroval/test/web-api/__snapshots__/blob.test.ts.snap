// Vitest Snapshot v1

exports[`Blob > serializeAsync > supports Blob 1`] = `"Promise.resolve(new Blob([new Uint8Array([72,101,108,108,111,32,87,111,114,108,100]).buffer],{type:\\"text/plain\\"}))"`;

exports[`Blob > toJSONAsync > supports Blob 1`] = `"{\\"t\\":{\\"t\\":12,\\"i\\":0,\\"f\\":{\\"t\\":23,\\"i\\":1,\\"c\\":\\"text/plain\\",\\"f\\":{\\"t\\":21,\\"i\\":2,\\"s\\":[72,101,108,108,111,32,87,111,114,108,100]}}},\\"r\\":0,\\"i\\":false,\\"f\\":16383,\\"m\\":[]}"`;
